src/main/java/edu/oregonstate/cs361/battleship/Main.java

```diff
@@ -801,64 +801,57 @@ public static boolean Hit(Point shipStart, Point shipEnd, Point shotPoint){
         return false; // points given are not horizontal or vertical and not valid, can't hit diagonally
     }
 
-    /*public static BattleshipModelUpdated Sink( Point shipStart, Point shipEnd, boolean player, BattleshipModelUpdated model ){
-        Point Shot = new Point();
+    public static BattleshipModelUpdated Sink( Point shipStart, Point shipEnd, boolean player, BattleshipModelUpdated model ){
 
         if( shipStart.getDown() == shipEnd.getDown() ){
             int y = shipStart.getDown();
 
             for ( int x = shipStart.getAcross(); x <= shipEnd.getAcross(); x++ ){
+                Point Shot = new Point();
                 Shot.setAcross(x);
                 Shot.setDown(y);
```

Comment:
But thanks for finding that error. If you set the Point again it will change the point that is in the array.